### PR TITLE
DAC: Add wrapper around RIOTs DAC-interface

### PR DESCRIPTION
**Pull request by Remmirad**
Adds a wrapper around RIOTS [dac-interface](https://doc.riot-os.org/group__drivers__periph__dac.html)

### CODE REVIEW

File: src/dac.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+use riot_sys::dac_t;
+
+#[derive(Debug)]
+pub struct DACLine {
+    line: dac_t,
+}
+
+#[derive(Debug)]
```
Comment (chrysn) [inline src/dac.rs:8]: This should better be `#[non_exhaustive]` so that we can add options later

File: src/dac.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    pub unsafe fn new_without_init(line: dac_t) -> Self {
+        DACLine { line }
+    }
+
+    /// Writes the given value to this [`DACLine`]
+    ///
+    /// The `value` is internally scaled to the underlying
+    /// dac device so that the maximum voltage output
+    /// is always equivalent to [`u16::MAX`]
+    pub fn set(&self, value: u16) {
```
Comment (chrysn) [inline src/dac.rs:47]: I'm conflicted about this and the power methods not taking a &mut.

Yes, RIOT provides the abstractions this way, but at the same time, picking the more Rust-idiomatic choice would help keep the Rust interface idiomatic if RIOT should find later that maybe working off the same initialized dac_t simultaneously in different threads is *not* such a good idea (but it just wasn't documented).

What are your thoughts on this?

**Review by chrysn — COMMENTED** [review]
Thanks for providing this PR, two comments/questions in-code.

File: src/dac.rs (hunk 2 above)
Comment (Remmirad) [reply src/dac.rs:47]: That's a good idea, just did not thought about it. I can't think of any justification for those _not_ to be `mut` in the Rust interface. If you really need to access it from different threads one should use safe Rust ways to do that. 
Ill change those to `mut` references.

Comment (Remmirad) [reply src/dac.rs]: Done

File: src/dac.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    Unknown,
+}
+
+impl DACLine {
+    /// Creates and intializes a new [`DACLine`] from the given
+    /// unitialized [`dac_t`].
+    ///
+    /// Returns an Error if the given line does not exist
+    /// on the board.
+    pub fn new(line: dac_t) -> Result<Self, DACError> {
```
Comment (chrysn) [inline src/dac.rs:23]: Would you mind trying this out with https://github.com/RIOT-OS/rust-riot-wrappers/issues/37#issuecomment-1430112835 ?

Comment (Remmirad) [reply src/dac.rs:23]: It seems that [`DAC_LINE`](https://doc.riot-os.org/group__drivers__periph__dac.html) is missing from `riot_sys/build.rs` which seems to be the equivalent for `*_DEV` in the DAC interface.

Comment (chrysn) [reply src/dac.rs:23]: Sorry, I missed that reply. A version of -sys that adds DAC_LINE is [being merged](https://github.com/RIOT-OS/rust-riot-sys/pull/22), and should thus be usable easily in a few mintues

Comment (Remmirad) [reply src/dac.rs]: Thank you, is now done in https://github.com/RIOT-OS/rust-riot-wrappers/pull/36/commits/25dead1549558fe43da4a4b02958c2a5d9df45a2

**Review by chrysn — APPROVED** [review]
Great, I think this is ready.

I've taken the liberty to apply (and force-push) some white space fixes right away -- things wouldn't have been easier had I asked you to apply that manually.

I've added a minimal test application that ensures this actually gets built, which is added on top of your commits.

**Comment by Remmirad** [comment]
Thank you :thumbsup:

**Pull request merged**